### ISSUE TITLE
Question about the mean-teacher setting

### ISSUE THREAD

**Issue by Hugo-cell111** [kunzhan/DSSN#3]
Hi! Have you done the ablation experiment of adopting mean-teacher to provide pseudo labels? In my opinion after various image and feature perturbations, it is no need to adopt MT, the model itself, with applying the classwise-aware threshold, can strong enough to dig out information and generate reliable pseudo labels. I think the performance boosting will be mild, am I right? Thanks!

**Comment by kunzhan** [comment]
We had tried this before, after all, Teacher does not have dropout operations, so its performance would be better. According to our observation, Teacher is more useful at the beginning of training.

**Comment by Hugo-cell111** [comment]
Is the difference between two settings mild or obvious?

**Comment by kunzhan** [comment]
one is model.eval, the other is model.train
train model has dropout while eval without dropout.

**Comment by Hugo-cell111** [comment]
Oh, I mean whether the difference is mild or obvious between two settings (mean teacher & a unified model like UniMatch). I know  there are two models in mean-teacher setting, one in model.eval() while the other in model.train()

**Comment by zhibotian** [comment]
I can't recall the exact improvement in outcomes because of time constraints, but I think the Teacher model is useful for model training, particularly in the early stages of training. 

 If you observe the training logs, the Teacher's test set results are superior to the Student's, and the primary purpose we employ the Mean-Teacher structure is that we hope Teacher generate better pseudo-labels in the subsequent iteration. 

You may also try running the code yourself. If you specify the right dataset path, our code can execute directly. Thank you.

**Comment by Hugo-cell111** [comment]
Thanks a lot!
